lib/iris/_concatenate.py

```diff
@@ -464,6 +465,12 @@ def match(self, other, error_on_mismatch):
             msgs.append(msg_template.format('Datatypes', '',
                                             self.data_type, other.data_type))
 
+        # Check _cell_measures_and_dims
+        if self._cell_measures_and_dims != other._cell_measures_and_dims:
```

Comment:
@marqh From a CF perspective I could see that for cell **methods** the order might be important to the provenance and history of the data i.e. a mean then a variance is different to variance then a mean.

If we have a sequence of cell **measures ** for a cube and our understanding is that order doesn't matter (I'm kinda guided by your interpretation and expertise in all things CF here) then I'd be in favour of imposing a sort order which you could base on the `CellMeasure._as_defn()`.

If you agree with that as an approach, the next decision is at which point do you sort? I'd suggest it's imposed during `Cube.add_cell_measure` i.e. something along the lines of this (which combines `Cube.add_cell_measure` and `Cube._add_unique_cell_measure` ...

```python
class Cube(CFVariableMixin):
    ....
    def add_cell_measure(self, cell_measure, data_dims=None):
        if self.cell_measures(cell_measure):
            raise ValueError('Duplicate cell measures are not permitted.')
        data_dims = self._check_multi_dim_metadata(cell_measure, data_dims)
        cm_and_dims = self._cell_measures_and_dims
        cm_and_dims.append([cell_measure, data_dims])
        cm_and_dims.sort(key=lambda (cm, dims): (cm._as_defn(), dims))
```